tests/BoltOn.Tests/Data/CosmosDb/CosmosDbRepositoryTests.cs

```diff
@@ -1,52 +1,145 @@
-		{
-			if (!IntegrationTestHelper.IsCosmosDbServer)
-				return;
-
-			// arrange
-			var id = Guid.Parse("56bbd30b-be53-4ec4-9e6b-8c2f4b6f8c71");
-			var student = new StudentFlattened { Id = id };
-
-			// act
-			await _sut.DeleteAsync(student, new RequestOptions { PartitionKey = new Microsoft.Azure.Documents.PartitionKey(2) });
-
-			// assert
-			var queryResult = _sut.GetById(id, new RequestOptions { PartitionKey = new Microsoft.Azure.Documents.PartitionKey(2) });
-			Assert.Null(queryResult);
-		}
-	}
+    [Collection("IntegrationTests")]
+    public class CosmosDbRepositoryTests : IDisposable
+    {
+        private readonly IRepository<StudentFlattened> _sut;
+
+        public CosmosDbRepositoryTests()
+        {
+            // this flag can be set to true for [few] tests. Running all the tests with this set to true might slow down.
+            IntegrationTestHelper.IsCosmosDbServer = true;
+            IntegrationTestHelper.IsSeedCosmosDbData = true;
+            var serviceCollection = new ServiceCollection();
+            serviceCollection
+                .BoltOn(options =>
+                {
+                    options
+                        .BoltOnCosmosDbModule();
+                });
+            var serviceProvider = serviceCollection.BuildServiceProvider();
+            serviceProvider.TightenBolts();
+            _sut = serviceProvider.GetService<IRepository<StudentFlattened>>();
+        }
+
+        [Fact, Trait("Category", "Integration")]
+        public async Task DeleteAsync_DeleteById_DeletesTheEntity()
+        {
+            if (!IntegrationTestHelper.IsCosmosDbServer)
+                return;
+
+            // arrange
+            var id = Guid.Parse("eda6ac19-0b7c-4698-a1f7-88279339d9ff");
+            var student = new StudentFlattened { Id = id };
+
+            // act
+            await _sut.DeleteAsync(student, new RequestOptions { PartitionKey = new PartitionKey(1) });
+
+            // assert
+            var queryResult = _sut.GetById(id, new RequestOptions { PartitionKey = new PartitionKey(1) });
+            Assert.Null(queryResult);
+        }
+
+        [Fact, Trait("Category", "Integration")]
+        public async Task GetByIdAsync_WhenRecordExists_ReturnsRecord()
+        {
+            // arrange
+            var id = Guid.Parse("eda6ac19-0b7c-4698-a1f7-88279339d9ff");
+
+            // act
+            var result = await _sut.GetByIdAsync(id, new RequestOptions { PartitionKey = new PartitionKey(1) });
+
+            // assert
+            Assert.NotNull(result);
+            Assert.Equal("John".ToLower(), result.FirstName.ToLower());
+        }
+
+        [Fact, Trait("Category", "Integration")]
+        public async Task GetAllAsync_WhenRecordsExist_ReturnsAllTheRecords()
+        {
+            // arrange
+
+            // act
+            var result = await _sut.GetAllAsync();
+
+            // assert
+            Assert.Single(result);
```

Comment:
assert the count, based on the seeded data... something like count > 2

Reply:
@gokulm : i was doing that earlier but asserting the count of the collection is giving warning with xunit and i googled online to see to avoid warning and they said if its one item do this, when we have to test other use cases we can change this.

Reply:
Assert.True(collection.Count > 2)